river/Keyboard.zig

```diff
@@ -82,6 +82,10 @@ fn handleKey(listener: *wl.Listener(*wlr.Keyboard.event.Key), event: *wlr.Keyboa
 
     self.seat.clearRepeatingMapping();
 
+    if (server.config.cursor_hide_when_typing == .enabled) {
+        self.seat.cursor.hide();
+    }
+
```

Comment:
This can cause the cursor to be hidden while cursor buttons are pressed. We should be able to fix this by moving the `self.pressed_count > 0` check from `getHideTimeout()` to `hide()`.

Reply:
Resolved it using the suggested solution. Hiding the cursor while typing is weird, i.e. hitting mod to move/resize a view will currently hide the cursor until the button is pressed which looks strange, maybe modifiers shouldn't hide the cursor.

Reply:
I agree, it would be better if modifiers didn't hide the cursor. Unfortunately `libxkbcommon` doesn't expose anything nice to check this in its public API afaict. It does however have this internal function:
https://github.com/xkbcommon/libxkbcommon/blob/a2507c0866543542e1bf310ca27695d7b33c83a1/src/keysym.c#L240-L248. Copying just the first part of that check (`(keysym >= XKB_KEY_Shift_L && keysym <= XKB_KEY_Hyper_R)`) should be sufficient for our use-case I believe.